eclair-core/src/main/scala/fr/acinq/eclair/payment/LocalPaymentHandler.scala

```diff
@@ -38,44 +36,30 @@ import scala.util.Try
   */
 class LocalPaymentHandler(nodeParams: NodeParams) extends Actor with ActorLogging {
 
-  import LocalPaymentHandler._
-
   implicit val ec: ExecutionContext = context.system.dispatcher
-  context.system.scheduler.schedule(10 minutes, 10 minutes)(self ! PurgeExpiredRequests)
-
-  override def receive: Receive = run(Map.empty)
+  lazy val paymentDb = nodeParams.db.payments
 
-  def run(hash2preimage: Map[ByteVector32, PendingPaymentRequest]): Receive = {
-
-    case PurgeExpiredRequests =>
-      context.become(run(hash2preimage.filterNot { case (_, pr) => hasExpired(pr) }))
+  override def receive: Receive = {
 
     case ReceivePayment(amount_opt, desc, expirySeconds_opt, extraHops, fallbackAddress_opt) =>
       Try {
-        if (hash2preimage.size > nodeParams.maxPendingPaymentRequests) {
+        if (paymentDb.listPendingPaymentRequests().size > nodeParams.maxPendingPaymentRequests) {
```

Comment:
This `Try` pattern isn't great. It was there before though.

Reply:
It's still useful, the PaymentRequest creation can go wrong and this allows to send a feedback back to the sender.

Reply:
This I think would be more scala-esque:

```scala
Try {
        val paymentPreimage = randomBytes32
        val paymentHash = Crypto.sha256(paymentPreimage)
        val expirySeconds = expirySeconds_opt.getOrElse(nodeParams.paymentRequestExpiry.toSeconds)
        val paymentRequest = PaymentRequest(nodeParams.chainHash, amount_opt, paymentHash, nodeParams.privateKey, desc, fallbackAddress_opt, expirySeconds = Some(expirySeconds), extraHops = extraHops)
        log.debug(s"generated payment request={} from amount={}", PaymentRequest.write(paymentRequest), amount_opt)
        paymentDb.addPaymentRequest(paymentRequest, paymentPreimage)
      } match {
        case Success(paymentRequest) => sender ! paymentRequest
        case Failure(t) => sender ! Status.Failure(t) 
      }
```